apps/antalmanac/src/components/RightPane/SectionTable/SectionTableBody.tsx

```diff
@@ -349,97 +362,119 @@ interface SectionTableBodyProps {
-                const courseStartTime = event.start.toString().split(' ')[4].slice(0, -3);
-                const courseEndTime = event.end.toString().split(' ')[4].slice(0, -3);
-                return { day: courseDay, startTime: courseStartTime, endTime: courseEndTime };
-            });
-
-            const coursePaneEvent = {
-                // If there already exists a more well-written way to translate secton.meetings days (string) into a number, LMK
-                // Converts SuTuTh -> [Su, Tu, Th] -> [0, 2, 4]
-                day: section.meetings[0].days.match(/[A-Z][a-z]*/g)?.map((day: string) => DAYS_TO_NUMS[day]),
-                startTime: '',
-                endTime: '',
-            };
-
-            const translatedTimeString = translateWebSOCTimeTo24HourTime(section);
-            if (translatedTimeString) {
-                coursePaneEvent.startTime = translatedTimeString.startTime;
-                coursePaneEvent.endTime = translatedTimeString.endTime;
-            }
+    // Because the idiot who set up state management didn't use an actual library solution,
+    // we need to attach ***memoized*** listeners to the store like bufoons.
 
-            for (const calendarEvent of calendarEventTimes) {
-                // Check if there is day overlap
-                if (!coursePaneEvent?.day?.includes(calendarEvent?.day)) {
-                    continue;
-                }
-
-                // Then, IF the course doesn't ( start AND end BEFORE) AND doesn't ( start AND end AFTER), it does conflict!
-                const happensBefore =
-                    coursePaneEvent.startTime <= calendarEvent.startTime &&
-                    coursePaneEvent.endTime <= calendarEvent.startTime;
-                const happensAfter =
-                    coursePaneEvent.startTime >= calendarEvent.endTime &&
-                    coursePaneEvent.endTime >= calendarEvent.endTime;
-
-                // If neither happensBefore or happensAfter is true, set scheduleConflict to true
-                if (!happensBefore && !happensAfter) {
-                    setScheduleConflict(true);
-                    return;
-                }
-            }
-            setScheduleConflict(false);
-            return;
+    useEffect(() => {
+        RightPaneStore.on('columnChange', handleColumnChange);
+        return () => {
+            RightPaneStore.removeListener('columnChange', handleColumnChange);
         };
+    }, [handleColumnChange]);
 
-        const updateCourseState = () => {
-            updateHighlight();
-            checkAndDisplayScheduleConflict();
+    useEffect(() => {
+        AppStore.on('addedCoursesChange', updateHighlight);
+        AppStore.on('currentScheduleIndexChange', updateHighlight);
+
+        return () => {
+            AppStore.removeListener('addedCoursesChange', updateHighlight);
+            AppStore.removeListener('currentScheduleIndexChange', updateHighlight);
         };
+    }, [updateHighlight]);
```

Comment:
For clarity, all event handlers are attached and removed in their own useEffect blocks